crates/analytics/src/refunds/distribution/sessionized_distribution/refund_error_message.rs

```diff
@@ -0,0 +1,177 @@
+{
+    async fn load_distribution(
+        &self,
+        distribution: &RefundDistributionBody,
+        dimensions: &[RefundDimensions],
+        auth: &AuthInfo,
+        filters: &RefundFilters,
+        granularity: &Option<Granularity>,
+        time_range: &TimeRange,
+        pool: &T,
+    ) -> MetricsResult<Vec<(RefundMetricsBucketIdentifier, RefundDistributionRow)>> {
+        let mut query_builder: QueryBuilder<T> =
+            QueryBuilder::new(AnalyticsCollection::RefundSessionized);
+
+        for dim in dimensions.iter() {
+            query_builder.add_select_column(dim).switch()?;
+        }
+
+        query_builder
+            .add_select_column(&distribution.distribution_for)
+            .switch()?;
+
+        query_builder
+            .add_select_column(Aggregate::Count {
+                field: None,
+                alias: Some("count"),
+            })
+            .switch()?;
+        query_builder
+            .add_select_column(Aggregate::Min {
+                field: "created_at",
+                alias: Some("start_bucket"),
+            })
+            .switch()?;
+        query_builder
+            .add_select_column(Aggregate::Max {
+                field: "created_at",
+                alias: Some("end_bucket"),
+            })
+            .switch()?;
+
+        filters.set_filter_clause(&mut query_builder).switch()?;
+
+        auth.set_filter_clause(&mut query_builder).switch()?;
+
+        time_range
+            .set_filter_clause(&mut query_builder)
+            .attach_printable("Error filtering time range")
+            .switch()?;
+
+        query_builder
+            .add_filter_clause(
+                RefundDimensions::RefundStatus,
+                storage_enums::RefundStatus::Failure,
+            )
+            .switch()?;
+
+        for dim in dimensions.iter() {
+            query_builder
+                .add_group_by_clause(dim)
+                .attach_printable("Error grouping by dimensions")
+                .switch()?;
+        }
+
+        query_builder
+            .add_group_by_clause(&distribution.distribution_for)
+            .attach_printable("Error grouping by distribution_for")
+            .switch()?;
+
+        if let Some(granularity) = granularity.as_ref() {
+            granularity
+                .set_group_by_clause(&mut query_builder)
+                .attach_printable("Error adding granularity")
+                .switch()?;
+        }
+
+        for dim in dimensions.iter() {
+            query_builder.add_outer_select_column(dim).switch()?;
+        }
+
+        query_builder
+            .add_outer_select_column(&distribution.distribution_for)
+            .switch()?;
+        query_builder.add_outer_select_column("count").switch()?;
+        query_builder
+            .add_outer_select_column("start_bucket")
+            .switch()?;
+        query_builder
+            .add_outer_select_column("end_bucket")
+            .switch()?;
```

Comment:
Consider using enums here (in the query builder method) instead of string literals, this can be very error-prone.
Of course, this can be taken up in a separate PR.

Reply:
Will take this up while refactoring in a separate PR.